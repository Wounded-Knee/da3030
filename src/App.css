/* Boss Screen
****************/
html.bossMode {
  opacity: 0.05;
  filter: grayscale(100%);
}
html.bossMode:hover {
  opacity: 1;
}
body {
  background: #ccc;
}

.App {
  text-align: center;
}

/* Z-Index
***************/
header #money,
header {
  z-index: 1;
}

/* Buttons
*************/
button {
  cursor: pointer;
  background: transparent;
  border: 0;
}

/* HEADER / FOOTER
******************/
header,
footer {
  background-color: #282c34;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

/* HEADER
******************/
header {
  position: fixed;
  left: 0;
  right: 0;
}

header .userSelector {
  font-size: 2em;
}

/* HEADER -> Nav Menu
***********************/
header > ul {
  list-style-type: none;
  padding: 0;
  margin: 0;
  margin-bottom: -2px;
  font-size: 40px;
}
header > ul li {
  position: relative;
  display: inline-block;
  margin: 0 1em;
}
header > ul li .count {
  position: absolute;
  bottom: 1em;
  font-weight: bold;
  display: block;
  width: 1.5em;
  height: 1.5em;
  color: #fff;
  background: #f00;
  border-radius: 2em;
  font-size: 0.25em;
  margin-right: -1em;
}
header > ul li a:hover,
header > ul li a {
  color: #8b92a0;
  text-decoration: none;
  border: 0.1em solid transparent;
}
header > ul li a:hover {
  color: #fff;
  border-bottom-color: #5c748c;
}
header > ul li a.active {
  font-weight: bold;
  border-bottom-color: #ccc;
}

/* HEADER -> Money Menu
************************/
header #money,
header #moneyMenu {
  position: absolute;
}
header #money {
  left: 0;
  top: 32px;
  transform: rotate(-25deg);
  background: none;
  border: 0;
  font-size: 38px;
  text-shadow: 2px 3px 9px black;
  line-height: 0em;
}
header #moneyMenu {
  display: none;
  left: 0;
  background: #282c34;
  color: #fff;
  margin-top: 10em;
  padding-top: 30em;
  width: 57px;
  font-size: 0.6em;
}
header #moneyMenu ul {
  padding: 0;
}
header #moneyMenu.open {
  display: block;
}
header #moneyMenu li {
  cursor: pointer;
  display: block;
}
header #moneyMenu li:hover {
  background: #000;
}
header #moneyMenu .logo {
  font-size: 13em;
}
header #moneyMenu .logo .d {
  color: #336699;
  font-size: 0.5em;
  font-weight: bold;
}
header #moneyMenu .logo .three {
  color: #fff18a;
}

/* HEADER -> User Selector
************************/
header select {
  position: absolute;
  right: 1em;
}

/* MAIN
*********/
main {
  padding-top: 5em;
}

/* HOME
************/
main ul {
  list-style-type: none;
  padding: 0;
  margin: auto;
  width: 500px;
}
main ul li {
  clear: both;
  margin: 0 1em;
}
main ul li a {
  color: #000;
  text-decoration: none;
}
main ul li a.active {
  font-weight: bold;
}
main span {
  font-size: 0.5em;
  margin: -1em 0 1em;
}

/* Node Components
*******************/
main button {
  cursor: pointer;
  background: transparent;
  border: 0;
  font-size: 15px;
}

.App-link {
  color: #09d3ac;
}

/* Node List
**************/
.nodeList {
  max-width: 500px;
  margin: auto;
}

/* Node
************/
article.node .controls button {
  font-size: 15px;
}

/* Node -> Controls -> Carousel
******************************/
.nodeList li .slick-slider,
article.node .slick-list {
  clear: both;
}

/* Node -> Controls -> Visibilities
************************************/
article.node .controls .author,
article.node .controls .exposure,
article.node .controls button {
  display: none;
}
article.node:hover .controls .author,
article.node:hover .controls .exposure,
article.node:hover .controls button {
  display: inline-block;
  opacity: 0.5;
}
article.node .controls .author:hover,
article.node .controls .exposure:hover,
article.node .controls button:hover {
  opacity: 1;
}

/* Node -> Speech Bubble
***************************/
article.node {
  position: relative;
  margin: 1em 5em;
  padding: 1em;
  border-radius: .4em;
  clear: both;
}

article.node a {
  color: #000;
  text-decoration: none;
}

article.node:after {
  content: '';
  position: absolute;
  top: 50%;
  width: 0;
  height: 0;
  border: 10px solid transparent;
  margin-top: -10px;
}

article.node:before {
  position: absolute;
  font-size: 3em;
  top: 0;
}

/* Node -> Speech Bubble: Left
******************************/
article.node:not(.author) {
  background: #fffdb8;
  text-align: left;
  float: left;
}

article.node:not(.author):after {
  left: 0;
  border-right-color: #fffdb8;
  border-left: 0;
  margin-left: -10px;
}

article.node:not(.author):before {
  left: -1.5em;
}
article.node:not(.author) .controls {
  float: right;
}

/* Node -> Speech Bubble: Left -> Trail Warden
***********************************************/
article.node.trailWarden {
  -webkit-box-shadow: 0px 0px 5px 0px rgba(255,255,255,0.75);
  -moz-box-shadow: 0px 0px 5px 0px rgba(255,255,255,0.75);
  box-shadow: 0px 0px 5px 0px rgba(255,255,255,0.75);
}

/* Node -> Speech Bubble: Right
*********************************/
article.node.author {
  background: #62bcff;
  text-align: right;
  float: right;
}

article.node.author:after {
  right: 0;
  border-left-color: #62bcff;
  border-right: 0;
  margin-right: -10px;
}

article.node.author:before {
  right: -1.5em;
}

article.node.author .controls {
  float: left;
}

article.node .react-tabs {
  clear: both;
}

.certificates > ul > li {
  opacity: 0.5;
  cursor: pointer;
  display: inline-block;
  font-size: 2em;
}
.node .certificates > ul > li:not(.selected) {
  color: transparent;
  background: #ccc;
  -webkit-background-clip: text;
  -moz-background-clip: text;
  background-clip: text;
}
.certificates > ul > li.selected,
.certificates > ul > li:hover {
  opacity: 1;
}
.certificates > ul > li.certified {
  color: #e0dfdc;
  background: #556677;
  letter-spacing: .1em;
  color: #e0dfdc;
  background: #556677;
}

/* Cloud
********/
.cloud {
  display: inline-block;
  position: relative;
}

/* FOOTER
******************/
footer {
  margin-top: 3em;
}

footer small {
  font-size: 0.5em;
}

/* Clearfix
************/
.clearfix::after {
  content: "";
  clear: both;
  display: table;
}
