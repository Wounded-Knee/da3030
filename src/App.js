import React from 'react';
import initialize from './initialize';
import UserSelector from './view/UserSelector';
import Profile from './view/Profile';
import CheatMenu from './view/CheatMenu';
import Node from './view/Node';
import {
  User,
  MODEL_TYPES,
} from './class/Models';
import {
  BrowserRouter as Router,
  Route,
  Switch,
  Redirect,
  NavLink,
} from 'react-router-dom';
import AnnuitC≈ìptisII from './class/AnnuitC≈ìptisII';
import Home from './view/Home';
import './App.css';

const activeClassName = 'active';

class App extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      renderAgain: 0,
    };

    this.annuitC≈ìptisII = new AnnuitC≈ìptisII({
      onChange: this.triggerRender.bind(this),
    });

    window.da = {
      ...window.da,
      app: this,
      ac2: this.annuitC≈ìptisII,
      refresh: this.triggerRender.bind(this),
      initialize: initialize.bind(this, this.annuitC≈ìptisII),
    };
  }

  redirect(id) {
    this.setState({
      ...this.state,
      redirectId: id
    });
  }

  triggerRender() {
    this.setState({
      ...this.state,
      renderAgain: this.state.renderAgain + 1,
    });
  }

  addNodePrompt(e) {
    const text = prompt('Node name?','');

    e.preventDefault();

    if (text) {
      this.annuitC≈ìptisII.create({
        text
      }, MODEL_TYPES.TEXT_NODE);
    }
  }

  setDocumentTitle(title) {
    document.title = title;
  }

  renderNotifications(notificationType) {
    const notifications = this.annuitC≈ìptisII.getByModelType(notificationType);
    const qty = notifications.length;
    return qty ? <span className="count">{ qty }</span> : null;
  }

  render() {
    const currentUser = this.annuitC≈ìptisII.getCurrentUser();
    const [ currentUserEmoji ] = currentUser.data.name;
    const currentUserName = currentUser.data.name.substring(2);
    const css = [
      ...this.annuitC≈ìptisII.filter(user => user.getModelType() === MODEL_TYPES.USER),
      User.getAnonymous(),
    ].map(
      user => {
        const [ emoji ] = user.data.name;
        return (`
          article.node.author_${emoji}:before {
            content: '${emoji}';
          }
        `)
      }
    ).join('\n');

    return (
      <div className="App">
        <style>{ css }</style>
        <Router>

          { this.state.redirectId ? <Redirect to={ `/node/${this.state.redirectId}` } /> : null }

          <header className="App-header">
            <CheatMenu da={ window.da } redirect={ this.redirect.bind(this) } annuitC≈ìptisII={ this.annuitC≈ìptisII } />
            <UserSelector annuitC≈ìptisII={ this.annuitC≈ìptisII } />
            <ul>
              <li title="Home" className="home">
                {/*
                  Home is where you see all the recent stuff contributed by your friends into
                  their own Home zone. We skim from your friends, and your followed entities,
                  the stuff they are posting for followers, it is all aggregated in HOME.
                */}
                { this.renderNotifications(MODEL_TYPES.HOME_NOTIFICATION) }
                <NavLink to="/" exact activeClassName={ activeClassName }>üè†</NavLink>
              </li>
              <li title="Clouds" className="clouds">
                {/*
                  Cloud is where you see a CLOUD of entities. These are your friends and your
                  followed entities, themselves. Here you can see the list of them, organized
                  geometrically. Your mom, your pastor, your buddy, your friend's dog.

                  Upon selecting an entity, you see only the stuff contributed by that entity,
                  but unlike with Home, you see ALL of it, and it's concentrated.
                */}
                { this.renderNotifications(MODEL_TYPES.CLOUD_NOTIFICATION) }
                <NavLink to="/clouds" exact activeClassName={ activeClassName }>‚òÅÔ∏è</NavLink>
              </li>
              <li title={ currentUserName } className="profile">
                {/*
                  Profile? Depends. Who's viewing it?
                    The owner of it
                      The owner sees everything he's ever posted.

                    Someone else
                      This is where your followed entities, and friends, see what you have posted
                      UP TO the boundary of the realm of privacy to which that friend / entity is party, according
                      to your own designation.
                */}
                { this.renderNotifications(MODEL_TYPES.USER_NOTIFICATION) }
                <NavLink to="/profile" exact activeClassName={ activeClassName }>{ currentUserEmoji }</NavLink>
              </li>
              <li title="Speak">
                <NavLink onClick={ this.addNodePrompt.bind(this) } to="/speak" exact activeClassName={ activeClassName }>üí¨</NavLink>
              </li>
            </ul>
          </header>

          <main>
            <Switch>

              <Route
                path="/"
                exact
                render={
                  props => <Home
                    {...props}
                    redirect={ this.redirect.bind(this) }
                    annuitC≈ìptisII={ this.annuitC≈ìptisII }
                  />
                }
              />

{/*              <Route
                path="/clouds"
                exact
                render={
                  props => <Clouds
                    {...props}
                    annuitC≈ìptisII={ this.annuitC≈ìptisII }
                  />
                }
              />
              <Route
                path="/cloud/:cloudId"
                render={
                  props => <Cloud
                    {...props}
                    annuitC≈ìptisII={ this.annuitC≈ìptisII }
                  />
                }
              />
*/}

              <Route
                path="/profile"
                exact
                render={
                  props => <Profile
                    {...props}
                    annuitC≈ìptisII={ this.annuitC≈ìptisII }
                  />
                }
              />

              <Route
                path="/profile/:userId"
                render={
                  props => <Profile
                    {...props}
                    annuitC≈ìptisII={ this.annuitC≈ìptisII }
                  />
                }
              />

              <Route
                path="/node/:nodeId"
                render={
                  props => <ul className="nodeList">
                    <li>
                      <Node
                        {...props}
                        setDocumentTitle={ this.setDocumentTitle.bind(this) }
                        redirect={ this.redirect.bind(this) }
                        annuitC≈ìptisII={ this.annuitC≈ìptisII }
                      />
                    </li>
                  </ul>
                }
              />

              <Redirect to="/" />

            </Switch>
          </main>

          <footer>
            <small>
              render #{ this.state.renderAgain }
            </small>
          </footer>
        </Router>
      </div>
    );
  }
}

export default App;
