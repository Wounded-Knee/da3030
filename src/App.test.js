import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';

const getUserByName = username => {
	return da.annuitCœptis.User.filter(
		user => user.data.name === username
	)[0];
}

it('renders without crashing', () => {
	const div = document.createElement('div');
	ReactDOM.render(<App />, div);
	ReactDOM.unmountComponentAtNode(div);
});

it('can add user', () => {
	da.annuitCœptis.User.create('Benjamin Franklin');
	expect(getUserByName('Benjamin Franklin').data).toMatchSnapshot();
})

it('can add second user', () => {
	da.annuitCœptis.User.create('John Locke');
	expect(getUserByName('John Locke').data).toMatchSnapshot();
})

it('can be user', () => {
	da.annuitCœptis.User.be(1);
	expect(da.annuitCœptis.getCurrentUser().data).toMatchSnapshot();
})

it('can add a node', () => {
	const newNode = da.annuitCœptis.Node.create('As people are walking all the time, in the same spot, a path appears.');
	expect(da.annuitCœptis.Node.getById(newNode.data.id).data).toMatchSnapshot();
})

it('can add a child node', () => {
	const parentNode = da.annuitCœptis.Node.getById(0);
	const childNode = da.annuitCœptis.Node.create('We are like chameleons, we take our hue and the color of our moral character, from those who are around us.', parentNode);
	expect(da.annuitCœptis.Node.getById(parentNode.data.id).children[0].data).toMatchSnapshot();
});

it('can track a visit to a node', () => {
	const trackedNode = da.annuitCœptis.Node.getById(0);
	const untrackedNode = da.annuitCœptis.Node.getById(1);
	const user = da.annuitCœptis.getCurrentUser();
	const track = da.annuitCœptis.Track.userAddTrack(trackedNode, user);

	expect(da.annuitCœptis.Track.userHasTrack(trackedNode.data.id, user)).toEqual(true);
	expect(da.annuitCœptis.Track.userHasTrack(untrackedNode.data.id, user)).toEqual(false);
});
