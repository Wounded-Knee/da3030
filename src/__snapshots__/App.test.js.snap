// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can add a child node 1`] = `
Object {
  "authorId": 1,
  "id": 1,
  "text": "We are like chameleons, we take our hue and the color of our moral character, from those who are around us.",
  "type": "node",
}
`;

exports[`can add a node 1`] = `
Object {
  "authorId": 1,
  "id": 0,
  "text": "As people are walking all the time, in the same spot, a path appears.",
  "type": "node",
}
`;

exports[`can add second user 1`] = `
Object {
  "authorId": undefined,
  "id": 1,
  "name": "John Locke",
  "type": "user",
}
`;

exports[`can add user 1`] = `
Object {
  "authorId": undefined,
  "id": 0,
  "name": "Benjamin Franklin",
  "type": "user",
}
`;

exports[`can be user 1`] = `
Object {
  "authorId": undefined,
  "id": 1,
  "name": "John Locke",
  "type": "user",
}
`;
